src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -681,6 +682,18 @@ private static IPackageFile FindFileInPackage(string filePath, IEnumerable<IPack
             return null;
         }
 
+        private void ValidateFilesUnique(IEnumerable<IPackageFile> files)
+        {
+            var set = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
zip files are case sensitive, so this check should be as well. 

We recently had a similar bug in our licenses code, https://github.com/NuGet/Home/issues/9817